scipy/stats/tests/test_stats.py

```diff
@@ -4808,118 +4808,125 @@ def test_location_sign(self, ksfunc, alternative,
         assert res.statistic_sign == ref_sign
 
 
-def test_ttest_rel():
-    # regression test
-    tr,pr = 0.81248591389165692, 0.41846234511362157
-    tpr = ([tr,-tr],[pr,pr])
-
-    rvs1 = np.linspace(1,100,100)
-    rvs2 = np.linspace(1.01,99.989,100)
-    rvs1_2D = np.array([np.linspace(1,100,100), np.linspace(1.01,99.989,100)])
-    rvs2_2D = np.array([np.linspace(1.01,99.989,100), np.linspace(1,100,100)])
-
-    t,p = stats.ttest_rel(rvs1, rvs2, axis=0)
-    assert_array_almost_equal([t,p],(tr,pr))
-    t,p = stats.ttest_rel(rvs1_2D.T, rvs2_2D.T, axis=0)
-    assert_array_almost_equal([t,p],tpr)
-    t,p = stats.ttest_rel(rvs1_2D, rvs2_2D, axis=1)
-    assert_array_almost_equal([t,p],tpr)
-
-    # test scalars
-    with suppress_warnings() as sup, \
-            np.errstate(invalid="ignore", divide="ignore"):
-        sup.filter(RuntimeWarning, "Degrees of freedom <= 0 for slice")
-        t, p = stats.ttest_rel(4., 3.)
-    assert_(np.isnan(t))
-    assert_(np.isnan(p))
+@pytest.mark.skip_xp_backends(cpu_only=True,
+                              reasons=['Uses NumPy for pvalue, CI'])
+@pytest.mark.usefixtures("skip_xp_backends")
+@array_api_compatible
+class TestTTestRel:
+    def test_regression(self, xp):
+        # regression test
+        tr, pr = 0.81248591389165692, 0.41846234511362157
+        tpr = xp.asarray([[tr, -tr], [pr, pr]])
+
+        rvs1 = xp.linspace(1, 100, 100)
+        rvs2 = xp.linspace(1.01, 99.989, 100)
+        rvs1_2D = xp.stack([xp.linspace(1, 100, 100), xp.linspace(1.01, 99.989, 100)])
+        rvs2_2D = xp.stack([xp.linspace(1.01, 99.989, 100), xp.linspace(1, 100, 100)])
+
+        t, p = stats.ttest_rel(rvs1, rvs2, axis=0)
+        xp_assert_close(xp.asarray([t, p]), xp.asarray([tr, pr]))
+        t, p = stats.ttest_rel(rvs1_2D.T, rvs2_2D.T, axis=0)
+        xp_assert_close(xp.asarray([t, p]), tpr)
+        t, p = stats.ttest_rel(rvs1_2D, rvs2_2D, axis=1)
+        xp_assert_close(xp.asarray([t, p]), tpr)
+
+        # test scalars
+        with suppress_warnings() as sup, \
+                np.errstate(invalid="ignore", divide="ignore"):
+            sup.filter(RuntimeWarning, "Degrees of freedom <= 0 for slice")
+            t, p = stats.ttest_rel(4., 3.)
+        assert xp.isnan(t)
```

Comment:
I also try to remember member to use `xp_assert_equal` for NaNs, too, because then it checks dtype and whether a NumPy value is a scalar or array.